ext/opentelemetry-ext-aiopg/src/opentelemetry/ext/aiopg/aiopg_integration.py

```diff
@@ -0,0 +1,144 @@
+import typing
+
+import wrapt
+from aiopg.utils import _ContextManager, _PoolAcquireContextManager
+
+from opentelemetry.ext.dbapi import DatabaseApiIntegration, TracedCursor
+from opentelemetry.trace import SpanKind
+from opentelemetry.trace.status import Status, StatusCanonicalCode
+
+
+# pylint: disable=abstract-method
+class AsyncProxyObject(wrapt.ObjectProxy):
+    def __aiter__(self):
+        return self.__wrapped__.__aiter__()
+
+    async def __anext__(self):
+        result = await self.__wrapped__.__anext__()
+        return result
+
+    async def __aenter__(self):
+        return await self.__wrapped__.__aenter__()
+
+    async def __aexit__(self, exc_type, exc_val, exc_tb):
+        return await self.__wrapped__.__aexit__(exc_type, exc_val, exc_tb)
+
+    def __await__(self):
+        return self.__wrapped__.__await__()
+
+
+class AiopgIntegration(DatabaseApiIntegration):
+    async def wrapped_connection(
+        self,
+        connect_method: typing.Callable[..., typing.Any],
```

Comment:
same here; if connect_method will always be based on aiopg, can we specialize this function?

Reply:
As far as i understand function `wrap_function_wrapper` needs that wrapper function receive wrapped function as first argument. Trying to set directly aiopg methods causes crossimport error